### ISSUE TITLE
Add publication year to dashboard

### ISSUE THREAD

**Issue by mietcls** [ugent-library/biblio-backoffice#914]
# Story

Add publication year to dashboard

## Context

In the old Biblio the reviewers had the option to filter on year in the dashboard. We have not implemented this yet, but the functionality is apparently more and more pressing to be able to do reviews for deadlines.

![Image](https://user-images.githubusercontent.com/1572885/207324463-cc9b1318-c7c0-466a-a51d-ee9ef7749195.png)

The dashboard story: https://github.com/ugent-library/biblio-backoffice/issues/517

## Current situation

![Image](https://user-images.githubusercontent.com/1572885/207366456-bb849ed8-c8f9-4a0b-901d-f08ce0970d13.png)

## Desired situation

- [x] For each dashboard
  - [x] Publications faculties
  - [x] Publications SOCs
- [x] For each table
  - [x] "Classification U"
  - [x] "Accepted"
- [x] These years:
  - [x] Selection for all publication years (every record) [DEFAULT]
  - [x] Selection for each year, descending (newest year first)

Out of scope:
  - Selection for missing publication years – this is not a filter in general yet

![Image](https://user-images.githubusercontent.com/1572885/207366090-590bba1f-4ca1-4d0e-b234-1b3090a3f228.png)

Example:
![Image](https://user-images.githubusercontent.com/1572885/207367339-59505cbe-da59-4d37-a422-4916cd06193e.png)

## Suggested implementation
Inside the card header:

```
<div class="card-header">
    <div class="bc-toolbar">
        <div class="bc-toolbar-left">
            <h3 class="card-title">Public with classification U</h3>
        </div>
        <div class="bc-toolbar-right text-align-right">
            <div>
                <label class="col-form-label sr-only">Select year</label>
                <select class="custom-select w-100">
                    <option>All years</option>
                    <option>Publication year missing</option>
                    <option>2023</option>
                    <option>2022</option>
                    <option>2021</option>
                    <option>2020</option>
                    <option>....</option>
                </select>
            </div>
        </div>
    </div>
</div>
```

**Comment by nicolasfranck** [comment]
@mietcls this is the "publication year", right?

**Comment by mietcls** [comment]
@nicolasfranck yes, edited ;-)

**Comment by nicolasfranck** [comment]
@mietcls @netsensei sigh, what I also see is that there is a select box for every table. Selecting that should update the tabel below it, right? But if I would implement this using the existing setup, a select would reload the page, after which you would have to scroll to the right table again..

**Comment by mietcls** [comment]
Providing this for "Public with classification U" is most important, as this one is used most. So the scrolling issue won't be that impactful. This feature is for librarians only. I can check if it's okay to only do this for the first table.

**Comment by nicolasfranck** [comment]
@mietcls the code for "publications with U" and "publication with accepted" are all in the same place. But no worries, I have a solution

**Comment by nicolasfranck** [comment]
Datasets do not have an attribute `year` (there is no facet for that reason)

**Comment by nicolasfranck** [comment]
@mietcls when I click on the year button, the table is reloaded in the background. But to an end user it is not so clear that the table is being reloaded, as there is no loading bar. Is there a way to put some half transparent layer over the table to make clear that it is loading something?

**Comment by mietcls** [comment]
Can you use this @nicolasfranck? https://github.com/ugent-library/biblio-backoffice/issues/610

**Comment by nicolasfranck** [comment]
@mietcls would be better to have something hovering over the existing table (which will be swapped out by the new table eventually). Will assign to a ticket once this is merged.
